include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
+            // TODO: It should be an exception
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to get Google Events');
             return false;
         }
 
-        $event_remote->setSummary($event_local->name);
-        $event_remote->setDescription($event_local->description);
-        $event_remote->setLocation($event_local->location);
-
-        $startDateTime = new Google_Service_Calendar_EventDateTime;
-        $startDateTime->setDateTime(date(DATE_ATOM, strtotime($event_local->date_start)));
-        $startDateTime->setTimeZone($this->timezone);
-        $event_remote->setStart($startDateTime);
-
-        $endDateTime = new Google_Service_Calendar_EventDateTime;
-        $endDateTime->setDateTime(date(DATE_ATOM, strtotime($event_local->date_end)));
-        $endDateTime->setTimeZone($this->timezone);
-        $event_remote->setEnd($endDateTime);
-
-        // We pull the existing extendedProperties, and change our values
-        // That way we don't mess with anything else that's using other values.
-        $extendedProperties = $event_remote->getExtendedProperties();
-
-        if (!empty($extendedProperties)) {
-            $private = $extendedProperties->getPrivate();
-        } else {
-            $extendedProperties = new Google_Service_Calendar_EventExtendedProperties;
-            $private = array();
-        }
-
-        $private['suitecrm_id'] = $event_local->id;
-        $private['suitecrm_type'] = $event_local->module_name;
-
-        $extendedProperties->setPrivate($private);
-        $event_remote->setExtendedProperties($extendedProperties);
-
-        // Copy over popup reminders
-        $event_local_id = $event_local->id;
-        $reminders_local = BeanFactory::getBean('Reminders')->get_full_list(
-            "",
-            "reminders.related_event_module = 'Meetings'" .
-            " AND reminders.related_event_module_id = '$event_local_id'" .
-            " AND popup = '1'"
-        );
+        foreach ($googleEvents as $gevent) {
+            $meeting = $this->getMeetingByEventId($gevent->getId());
+            $action = $this->pushPullSkip($meeting, $gevent);
+            $actionResult = $this->doAction($action, $meeting, $gevent);
 
-        if ($reminders_local) {
-            $reminders_remote = new Google_Service_Calendar_EventReminders;
-            $reminders_remote->setUseDefault(false);
-            $reminders_array = array();
-            foreach ($reminders_local as $reminder_local) {
-                $reminder_remote = new Google_Service_Calendar_EventReminder;
-                $reminder_remote->setMethod('popup');
-                $reminder_remote->setMinutes($reminder_local->timer_popup / 60);
-                $reminders_array[] = $reminder_remote;
+            if (!$actionResult) {
+                // TODO: Inform the caller about this problem (exception?)
```

Comment:
@gymad 
Cruft. doAction() will throw an exception on error. This will never get triggered.

Reply:
fixed